### ISSUE TITLE
Specify path for object refinement

### ISSUE THREAD

**Issue by ozimos** [vriad/zod#69]
refine is the recommended way for validating related properties on an object
https://github.com/vriad/zod/issues/61
Consider this schema

```
const validationSchema = z
  .object({
    firstName: z.string().optional(),
    lastName: z.string().optional(),
    email: z.string().email(),
    password: z.string(),
    confirmPassword: z.string(),
  })
  .refine((data) => data.password === data.confirmPassword, 'Both password and confirmation must match')

```
with this input data 

```
{
  firstName: 'zod',
  lastName: '',
  email: 'theba@zod.c',
  password: 'thetetathea',
  confirmPassword: 'thethtbet',
}
```

we get this error

```
[
  {
    message: 'Both password and confirmation must match',
    path: [],
  },
]
```

The path is empty. Ideally, the path should contain "confirmPassword" so that the error can be easily displayed on the UI

**Comment by vriad** [comment]
This is the desired default behavior. The custom check is associated with the entire object schema. There's no way for Zod to know that the error should be associated with `confirmPassword` in the code you provided.

Though there should be a way to support your use case. My proposal: the second argument to `.refine` can also accept an object, including an optional `path` key where you can specify the error path you'd like to be associated with the error. 

Something like this:

```ts
const validationSchema = z
  .object({
    firstName: z.string().optional(),
    lastName: z.string().optional(),
    email: z.string().email(),
    password: z.string(),
    confirmPassword: z.string(),
  })
  .refine((data) => data.password === data.confirmPassword, {
    path: ['confirmPassword'],
    message: 'Both password and confirmation must match'
  })
```

Which results in:

```ts
validationSchema.parse({
  firstName: 'zod',
  lastName: '',
  email: 'theba@zod.c',
  password: 'thetetathea',
  confirmPassword: 'thethtbet',
});

/*
[
  {
    message: 'Both password and confirmation must match',
    path: ['confirmPassword'],
  },
]
*/
```
Would that work?

**Comment by ozimos** [comment]
@vriad  Yes that would be awesome

**Comment by ozimos** [comment]
Thanks for this great library.

**Comment by vriad** [comment]
@ozimos This is now implemented in zod@1.8+

This exact use case is also documented as an example in the README: https://github.com/vriad/zod#custom-validation

**Comment by vriad** [comment]
@rossholdway @etienne-dldc Tagging for interest.
